### DISCUSSION TITLE
hold time violation on nangate45/swerv

### DISCUSSION THREAD

**Discussion by ttok9** [The-OpenROAD-Project/OpenROAD-flow-scripts#1280] [Q&A]
hi, everyone. I try to make nangate45/swerv.
I defined parameters like this on [config.mk](http://config.mk/)
export CAP_MARGIN = 60
export SLEW_MARGIN = 40
export PL_RESIZER_HOLD_MAX_BUFFER_PERCENT = 70
export PL_RESIZER_HOLD_SLACK_MARGIN = 0.8
export GLB_RESIZER_HOLD_MAX_BUFFER_PERCENT = 70
export GLB_RESIZER_HOLD_SLACK_MARGIN = 0.8
but there are hold time violation on final result.

![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/89254811/d8a9af1c-7a63-4bda-bfa9-2b04b5355a10)
This is default config.mk version hold violation 
![image (1)](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/89254811/71c72c77-24ca-4668-9108-88f070b3ec55)
This is modified version.
I modified config.mk for fix hold violation and slew, cap violation..
How can I fix hold violation? Thanks for any help!

**Comment by vijayank88** [comment]
Follow instruction and attach `make place_issue` test case here: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/issues/new?assignees=&labels=&projects=&template=bug_report_with_orfs.yml

**Reply by ttok9** [reply]
<img width="524" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/89254811/8556d3df-78f9-43bf-93cd-c61c18dcec12">
make place_issue seems to not work.. How can I make it?

**Reply by vijayank88** [reply]
ok. its nangate45/swerv design? Have you changed anything wrt., source files/configuration?

**Reply by ttok9** [reply]
Oh I made cts_issue and place_issue. Thank you! But tar size is too big to upload there..(389mb) I changed only config.mk(CAP margin, etc) and constraint file(clk speed)!

**Reply by ttok9** [reply]
export CAP_MARGIN = 60
export SLEW_MARGIN = 40
export PL_RESIZER_HOLD_MAX_BUFFER_PERCENT = 70
export PL_RESIZER_HOLD_SLACK_MARGIN = 0.8
export GLB_RESIZER_HOLD_MAX_BUFFER_PERCENT = 70
export GLB_RESIZER_HOLD_SLACK_MARGIN = 0.8
config.mk
and clk_period 400 on constraint.sdc

**Comment by vvbandeira** [comment]
@ttok9
This design is expected to have some hold violations with the default values, see here:
https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/e8bb32499b8fceb3f194b12f42cc03fce261bd95/flow/designs/nangate45/swerv/metadata-base-ok.json#L177

Please file an Issue (not a discussion) and fill in all relevant and required fields in the issue form.

**Comment by maliberty** [comment]
You are using OL variables in ORFS where they will have no effect.  There is already a nangate45/swerv (and swerv_wrapper).  What is the goal here?
